faunadb-scala/src/main/scala/faunadb/streaming/BodyValueFlowProcessor.scala

```diff
@@ -0,0 +1,82 @@
+package faunadb.streaming
+
+import java.nio.ByteBuffer
+import java.util
+import java.util.concurrent.{Flow, SubmissionPublisher}
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import faunadb.errors.PermissionDeniedException
+import faunadb.values.{StringV, VSuccess, Value}
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConverters.asScalaIteratorConverter
+import scala.util.Try
+
+private [faunadb] class BodyValueFlowProcessor(syncLastTxnTime: Long => Unit) extends SubmissionPublisher[Value] with Flow.Processor[util.List[ByteBuffer], Value] {
+  private val log = LoggerFactory.getLogger(getClass)
+  private val json = new ObjectMapper
+  private var subscription: Flow.Subscription = _
+  private var subscriber: Flow.Subscriber[_ >: Value] = _
+
+  // We do not request data from the publisher until we have one subscriber
+  override def subscribe(subscriber: Flow.Subscriber[_ >: Value]): Unit = {
+    if (this.subscriber == null) {
+      this.subscriber = subscriber
+      super.subscribe(subscriber)
+      requestOne()
+    } else
+      throw new IllegalStateException("BodyValueFlowProcessor can have only one subscriber")
+  }
+
+  override def onSubscribe(subscription: Flow.Subscription): Unit =
+    this.subscription = subscription
+
+  override def onNext(item: util.List[ByteBuffer]): Unit = {
+    import java.nio.charset.StandardCharsets
+    val text = item.iterator().asScala.map(StandardCharsets.US_ASCII.decode(_).toString).mkString("")
```

Comment:
Perhaps a string builder would be more efficient than the map + mkStrign pattern?

Reply:
It is possible but I am not sure it would be really significant TBH.

Here we are mapping lazily over an `Iterator` and then consuming it via `mkString` which uses a `StringBuilder` internally.

I assume writing the `for loop` ourselves will only save the cost of the `Iterator`.

Do you have something specific in mind :)

Reply:
You're right. I missed the iterator call. BTW: server encoding is UTF-8.